versions/schematic-2.md

```diff
@@ -0,0 +1,235 @@
+
+Resources identified by a resource location have an identifier string which is made up of two sections, the domain and the name, written as `domain:name`. If the domain is not specified then it is assumed to be `minecraft`. For example `planks` would identify the same resource as `minecraft:planks` as the `minecraft:` prefix is implicit.
+
+### Schema
+
+#### <a name="schematicObject"></a>Schematic Object
+
+This is the nameless root object of the schematic. It shall be a [NBT](#formFile) tag compound.
+
+##### Fields
+
+Field Name | Type | Description
+---|:---:|---
+<a name="schematicVersion"></a>Version | `NBT int` | **Required.** Specified the format version being used. It may be used to provide validation and auto-conversion from older versions. The current version is `2`. This shall be saved as [NBT](#formFile) tag int.
+<a name="schematicContentVersion"></a>ContentVersion | `NBT int` | **Required.** A version identifier for the contents. Representing the minecraft version, when serialized. Used for providing backwards compatibility. [Values are specified by Mojang](https://minecraft-de.gamepedia.com/Versionen#Version-IDs). Mojangs data converter for older versions will need this value for proper conversion. This shall be saved as [NBT](#formFile) tag int.
+<a name="schematicMetadata"></a>Metadata | [Metadata Object](#metadataObject) as `NBT compound` | **Optional.** Provides metadata about the schematic. This shall be saved as [NBT](#formFile) tag compound.
+<a name="schematicWidth"></a>Width | `NBT int` | **Required.** Specifies the width (the amount of blocks along the X-axis) of the schematic, must be interpreted as unsigned. This shall be saved as [NBT](#formFile) tag int.
+<a name="schematicHeight"></a>Height | `NBT int` | **Required.** Specifies the height (the amount of blocks along the Y-axis) of the schematic, must be interpreted as unsigned. This shall be saved as [NBT](#formFile) tag int.
+<a name="schematicLength"></a>Length | `NBT int` | **Required.** Specifies the length (the amount of blocks along the Z-axis) of the schematic, must be interpreted as unsigned. This shall be saved as [NBT](#formFile) tag int.
+<a name="schematicOffset"></a>Offset | `NBT int array`[3] | **Optional.** Specifies the relative offset of the schematic. This value is relative to the most minimal point in the schematics area. If present, this offset shall be added to all blocks, tiles and entities when pasting it into the world (may be negative). This shall be saved as [NBT](#formFile) tag int array.
+<a name="schematicPaletteMax"></a>PaletteMax | `NBT byte` | **Required.** Specifies the size of the block palette index in number of bytes needed for the maximum palette index. Implementations must use this to determine, which index format will be needed. The palette index may fit within a datatype smaller than a 32-bit integer or even a longer datatype like 64-bit integer. This shall be saved as [NBT](#formFile) tag byte.
```

Comment:
Check the chapter "Functional changes", there is a point, explaining:
"Changed PaletteMax from integer to byte, because 8 would be the maximum possible value for Java implementations (64-bit long)"